frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -1351,6 +1353,14 @@ class Worksheet extends React.Component {
         return count;
     }
 
+    handleDateFormat(date) {
+        // Append 'Z' to indicate the time is UTC time.
+        if (date && date[date.length - 1] !== 'Z') {
```

Comment:
Is there a chance the date will sometimes have Z and sometimes not Z at the end? If not, we can just remove this if statement and always append "Z".

Reply:
Sometimes the date might be undefined. So I keep the check for `if(date)` here.